docs/book/src/clusterctl/commands/alpha-topology-plan.md

```diff
@@ -1,3 +1,172 @@
 # clusterctl alpha topology plan
 
-**Under Construction**
+The `clusterctl alpha topology plan` command can be used to dry run the topology reconciler using an input file.
+
+The input file could contain a new/modified Cluster, a new/modified ClusterClass and/or new/modified templates.
+
+The dry run output would list the Clusters and ClusterClasses that would be affected by the input. The output also lists the
+objects that will be created, updated and deleted of a target cluster.
+
+This command can be used with and without a management cluster. In case the command is used without a management cluster the input should have
+all the objects needed to perform the dry run. In case the command is run with a management cluster the dry run operation will look for objects 
+in the real cluster as needed.
+
+```shell
+clusterctl alpha topology plan -f input.yaml -o output/
+```
+
+> Note: This command can be used with or without a management cluster. In case the command is used without a management cluster the input should have
+> all the objects needed to perform the dry run. In case the command is run with a management cluster the dry run operation will look for objects 
+> in the management cluster as needed.
+
+## Dry run
+
+The operation is composed of the following steps:
+* Set the namespace on objects in the input with missing namespace.
+* Run the Defaulting and Validation webhooks on the Cluster and ClusterClass objects in the input.
+* Dry run the topology reconciler on the target cluster.
+* Capture all changes observed during reconciliation.
+
+<aside class="note">
+
+<h1>Set missing namespace</h1>
+
+The namespace used for objects with missing namespace follows this precedence order:
+* Value passed using `--namespace`
+* Current namespace defined in kubeconfig
+* Value `default`
+
+</aside>
+
+<aside class="note warning">
+
+<h1>Defaulting and Validation of objects</h1>
+
+Defaulting and validation is only performed on the Cluster and ClusterClass objects in the input. Objects of all other kinds in the input should 
+be fully valid and any required default values should be set. Missing values will not be defaulted.
+
+> **Note on Validation**: If the command is used with a real management cluster, objects from the management cluster will be used to perform 
+> `ValidateUpdate` appropriately.
+
+</aside>
+
+## Input
+
+### `--file`, `-f` (REQUIRED)
```

Comment:
Should we mention that there could be multiple?
Absolutely up to you, we don't have to.

Reply:
I am inclined to keeping it simple here. Users who might be interested in using multiple files can quickly figure out that it is supported by running `clusterctl alpha topology plan --help`. The flag will show up as `-f, --file stringArray`.